MSMAll/MSMAllPipeline.sh

```diff
@@ -224,11 +208,23 @@ else
 		--output-fmri-name="${OutputfMRIName}" \
 		--fmri-proc-string="${fMRIProcSTRING}" \
 		--output-proc-string="${output_proc_string}" \
-		--demean="${Demean}" \
-		--variance-normalization="${VarianceNormalization}" \
-		--compute-variance-normalization="${ComputeVarianceNormalization}" \
-		--revert-bias-field="${RevertBiasField}" \
+		--start-frame="${StartFrame}" \
+		--end-frame="${EndFrame}" \
 		--matlab-run-mode="${MatlabRunMode}"
+		
+	# TODO: the current save and load for new OutputfMRIName might cause trouble in parallel mode
+	OutputfMRINameOverride=$(cat "/tmp/OutputfMRIName.txt")
+	rm /tmp/OutputfMRIName.txt
```

Comment:
This can be improved if I separate the part of frame checking and duration computation from `SingleSubjectConcat.sh` in the main script `MSMAllPipeline.sh`. Otherwise, @coalsont do you have any thought on this potential issue?

Reply:
I strongly suggest that the modified OutputfMRIName be generated in advance and passed into `SingleSubjectConcat.sh`, rather than coding a special naming convention into the internal script and trying to pass it back out.  I would generate the modified name in the launch script, and pass it to MSMAllPipeline.sh (assuming it doesn't need to actually use files from a previous concatenation), and not have any of this naming logic internal to the pipelines.

Reply:
My suggestion was not to programmatically modify OutputfMRIName, but to physically change the name when you run the code, along with specifying the optional arguments controlling the frame range e.g.,:

1) OutputfMRIName=rfMRI_REST; OutputRegName=MSMAll (all resting state fMRI data).
2) OutputfMRIName=rfMRI_REST_5min_1To100; OutputRegName=MSMAll_5min_1To100 (first ~1.25mins of each resting state fMRI run).
3) OutputfMRIName=fMRI_REST_TASK_5min_1To100; OutputRegName=MSMAll_REST_TASK_5min_1To100 (first ~1.25mins of each resting state and task fMRI run).

Reply:
I found one advantage to programmatically modify the name is that it can automatically calculate the duration with format such as "5mins36secs", "40secs". It won't be a problem if we intentionally omit the seconds part in the file names, or users are fully aware of how to calculate the duration.

Reply:
That looks nice actually.  If you want to do it that way, make sure that it appends to both OutputfMRIName and OutputRegName.

Reply:
I would vote for predictable filenames rather than adding nontrivial-to-calculate things to the filename.  Outputting the duration to the logfile might be nice, but having it in the filename itself sounds like a pain to run later scripts on.

Reply:
Tim this is showing up in pre-specified variable names that we have complete freedom over.  This won't mess up downstream scripts.

Reply:
Well, that was my original suggestion.

Reply:
The latest commit doesn't use programmed filenames anymore.